conformance/lang/expressions/multiplicative-expr/remainder.balt

```diff
@@ -0,0 +1,4348 @@
+}
+
+Test-Case: output
+Description: Test remainder operation for different decimal values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
+
+public function main() {
+     decimal a = 25.E1742d;
+     decimal b = 0.0d;
+     decimal c = a % b;
+     io:println(c); // @output NaN
+     io:println(25.E1742d % 0.0d); // @output NaN
+
+     a = 25.E17d;
+     b = 25.E8632d;
+     c = a % b;
+     io:println(c); // @output 2.5E18
+     io:println(25.E17d % 25.E8632d); // @output 2.5E18
+
+     a = 25.E17d;
+     b = 25.E17d;
+     c = a % b;
+     io:println(c); // @output 0
+     io:println(25.E17d % 25.E17d); // @output 0
+
+     a = -25.1E17d;
+     b = 25.1E17d;
+     c = a % b;
+     io:println(c); // @output 0
+     io:println(-25.1E17d % 25.1E17d); // @output 0
+
+     a = 25.E17d;
+     b = -25.E17d;
+     c = a % b;
+     io:println(c); // @output 0.0
+     io:println(25.E17d % -25.E17d); // @output 0.0
+
+     a = 25.78d;
+     b = -5.34d;
+     c = a % b;
+     io:println(c); // @output 4.42
+     io:println(25.78d % -5.34d); // @output 4.42
+
+     a = -25.78d;
+     b = 5.34d;
+     c = a % b;
+     io:println(c); // @output -4.42
+     io:println(-25.78d % 5.34d); // @output -4.42
+
+     a = 0.0d;
+     b = 0.0d;
+     c = a % b;
+     io:println(c); // @output NaN
+     io:println(0.0d % 0.0d); // @output NaN
+
+     a = -0.0d;
+     b = 0.0d;
+     c = a % b;
+     io:println(c); // @output NaN
+     io:println(-0.0d % 0.0d); // @output NaN
+}
+
+Test-Case: output
+Description: Test remainder operation for different nillable decimal values.
+Labels: multiplicative-expr, numeric-literal, float-point-literal
+
+public function main() {
+     decimal? a = 25.E1742d;
+     decimal? b = 0.0d;
+     decimal? c = a % b;
+     io:println(c); // @output NaN
+
+     a = 25.E17d;
+     b = 25.E8632d;
+     c = a % b;
+     io:println(c); // @output 2.5E18
+
+     a = 25.E17d;
+     b = 25.E17d;
+     c = a % b;
+     io:println(c); // @output 0
+
+     a = -25.1E17d;
+     b = 25.1E17d;
+     c = a % b;
+     io:println(c); // @output 0
+
+     a = 25.E17d;
+     b = -25.E17d;
+     c = a % b;
+     io:println(c); // @output 0.0
+
+     a = 25.78d;
+     b = -5.34d;
+     c = a % b;
+     io:println(c); // @output 4.42
+
+     a = -25.78d;
+     b = 5.34d;
+     c = a % b;
+     io:println(c); // @output -4.42
+
+     a = 0.0d;
+     b = 0.0d;
+     c = a % b;
+     io:println(c); // @output NaN
+
+     a = -0.0d;
+     b = 0.0d;
+     c = a % b;
+     io:println(c); // @output NaN
+
+     a = 25.78d;
+     b = ();
+     c = a % b;
+     io:println(c.toBalString()); // @output ()
+
+     a = ();
+     b = 5.34d;
+     c = a % b;
+     io:println(c.toBalString()); // @output ()
+}
+
+
+Test-Case: output
+Description: Test remainder operation for user-defined subtypes of float.
+Labels: multiplicative-expr, numeric-literal, float-literal, module-type-defn
+
+type Floats -2.0f|-1.0f|0.0f|1.0f|2.0f|-0.0f;
+
```

Comment:
Let's add some more tests for remainder with user-defined types of float and decimal similar to what we did for the division. Let's test with nilable user-defined subtypes as well.